openage/convert/tool/subtool/version_select.py

```diff
@@ -63,3 +63,23 @@ def get_game_version(srcdir, avail_game_eds, avail_game_exps):
             info(" * %s", expansion.expansion_name)
```

Comment:
v 

[ ค้นหา ] [ txt | html | pdf | bibtex ] [ Tracker ] [ WG ] [ Email ] [ Diff1 ] [ Diff2 ] [ Nits ] จาก: draft-lim-mpeg4-mime-03 มาตรฐานที่เสนอ อัปเดตโดย: 6381 Errata มีอยู่

คณะทำงานเครือข่าย Y. Lim ขอความคิดเห็น: 4337 net&tv Inc. หมวดหมู่: Standards Track D. Singer Apple Computer มีนาคม 2549 การลงทะเบียนประเภท MIME สำหรับ MPEG-4 สถานะของบันทึกนี้ เอกสารนี้ระบุโปรโตคอลติดตามมาตรฐานอินเทอร์เน็ตสำหรับ ชุมชนอินเทอร์เน็ตและขออภิปรายและข้อเสนอแนะสำหรับ การปรับปรุง โปรดดูฉบับปัจจุบันของ "อินเทอร์เน็ต มาตรฐานโปรโตคอลอย่างเป็นทางการ" (STD 1) สำหรับสถานะมาตรฐาน และสถานะของโปรโตคอลนี้ การแจกจ่ายบันทึกช่วยจำนี้ไม่จำกัดจำนวน ประกาศเกี่ยวกับลิขสิทธิ์ ลิขสิทธิ์ (C) สังคมอินเทอร์เน็ต (2006) เชิงนามธรรม เอกสารนี้กำหนดประเภท MIME มาตรฐานที่เกี่ยวข้องกับ MP4 ไฟล์. นอกจากนี้ยังแนะนำให้ใช้ประเภท MIME ที่ลงทะเบียนตาม ประเภทของเนื้อหา สารบัญ 1 . บทนำ ................................................. ... 2 2 . การเลือกประเภท MIME สำหรับไฟล์ MP4 ........................... 3 3 . ข้อควรพิจารณาของ IANA ................................................. 3 3.1 . ไฟล์ MP4 ................................................ ... 4 3.2 . ไฟล์ MP4 พร้อมเสียง แต่ไม่มีการนำเสนอภาพ ........ 5 3.3. ไฟล์ MP4 ที่มี MPEG-4 System Stream และไม่ใช่ การนำเสนอด้วยภาพและเสียง ................................. 6 3.4 . Initial Object Descriptor (IOD) ในรูปแบบไบนารี ........... 7 3.5 . Initial Object Descriptor (IOD) ในรูปแบบข้อความ .......... 8 4 . ข้อควรพิจารณาด้านความปลอดภัย ................................. 9 5 . รับทราบ ................................................ 9 6 . การอ้างอิงเชิงบรรทัดฐาน ............................................ 9 Lim & Singer Standards Track [หน้า 1] RFC 4337 ประเภท MIME MPEG-4 มีนาคม 2549 1 . บทนำ เอกสารนี้อธิบายคำจำกัดความมาตรฐานของประเภท MIME ที่เกี่ยวข้องกับไฟล์ MP4 และแนวทางการใช้งาน MPEG-4 (ISO/IEC 14496) เป็นมาตรฐานที่ออกแบบมาสำหรับการแสดงแทน และการส่งข้อมูลมัลติมีเดียผ่านการขนส่งที่หลากหลาย โปรโตคอล [ 1 ] ประกอบด้วยการจัดการฉากแบบโต้ตอบและภาพ และการแสดงเสียงตลอดจนการทำงานของระบบเช่น มัลติเพล็กซ์ การซิงโครไนซ์ และเฟรมเวิร์กตัวอธิบายอ็อบเจ็กต์ [ 2 ]. วิธีการทางประวัติศาสตร์สำหรับข้อมูล MPEG ได้รับการประกาศภายใต้ "วิดีโอ" และแนวทางนี้เป็นไปตาม ISO/IEC 14496 ใน นอกจากนี้ MIME บางประเภทถูกกำหนดภายใต้ "เสียง" และ "แอปพลิเคชัน" สำหรับลำธารที่ไม่มีการนำเสนอด้วยภาพ การแก้ไข 1 ของมาตรฐาน ISO/IEC 14496 (หรือที่เรียกว่าเวอร์ชัน 2) แนะนำไฟล์ประเภทมาตรฐานที่เรียกว่าไฟล์ MP4 สำหรับการห่อหุ้ม ข้อมูล ISO/IEC 14496 ขณะนี้มีการระบุแยกต่างหากเป็นไฟล์ MP4 รูปแบบ [ 4 ] ซึ่งจะขึ้นอยู่กับรูปแบบไฟล์สื่อ ISO พื้นฐาน [ 3 ]. ข้อกำหนดแยกต่างหาก [ 5 ] ครอบคลุมการจัดเก็บ Advanced การเข้ารหัสวิดีโอ (AVC) (เรียกอีกอย่างว่า H.264) [ 6 ] วัสดุในไฟล์ตาม ในรูปแบบไฟล์สื่อฐาน ISO ไฟล์ประเภท MP4 สามารถใช้ได้ใน หลายวิธี บางทีสิ่งสำคัญที่สุดคือการใช้เป็น รูปแบบการแลกเปลี่ยนข้อมูล ISO/IEC 14496 เป็นเนื้อหาดาวน์โหลด รูปแบบและเป็นรูปแบบที่อ่านโดยเซิร์ฟเวอร์สื่อสตรีมมิ่ง การใช้งานสองครั้งแรกนี้จะอำนวยความสะดวกอย่างมากหากมี ประเภท MIME มาตรฐานสำหรับให้บริการไฟล์เหล่านี้ (เช่น ผ่าน HTTP) มาตรฐาน ISO/IEC 14496 นั้นกว้าง ดังนั้นประเภทของข้อมูล ที่อาจอยู่ในไฟล์ดังกล่าวได้อาจแตกต่างกันไป สั้นๆ ง่ายๆ บีบอัด วิดีโอและเสียง (โดยใช้อัลกอริธึมการบีบอัดที่แตกต่างกันจำนวนหนึ่ง) สามารถรวม; ข้อมูลฉากแบบโต้ตอบ ข้อมูลเมตาเกี่ยวกับ การนำเสนอ; การอ้างอิงถึงกระแสสื่อ ISO/IEC 14496 ภายนอก ไฟล์และอื่นๆ ใช้ MIME ระดับบนสุดที่แตกต่างกันเพื่อระบุ ประเภทของเนื้อหาในไฟล์ Lim & Singer Standards Track [หน้า 2] RFC 4337 ประเภท MIME MPEG-4 มีนาคม 2549 2 . การเลือกประเภท MIME สำหรับไฟล์ MP4 ประเภท MIME ที่จะกำหนดให้กับไฟล์ MP4 จะถูกเลือกตาม เนื้อหา. แนวทางพื้นฐานสำหรับการเลือกประเภท MIME มีดังนี้ ดังนี้: ก) ถ้าไฟล์ไม่มีการนำเสนอทั้งภาพและเสียง แต่ เท่านั้น ตัวอย่างเช่น MPEG-J หรือ MPEG-7 ให้ใช้ application/mp4; b) สำหรับไฟล์อื่นๆ ทั้งหมด รวมถึงไฟล์ที่มี MPEG-J เป็นต้น ใน นอกจากการสตรีมวิดีโอหรือเสียงแล้ว ควรใช้วิดีโอ/mp4 อย่างไรก็ตาม: ค) สำหรับไฟล์ที่มีเสียงแต่ไม่มีด้านภาพ รวมทั้งที่ มี MPEG-J เป็นต้น นอกเหนือจากสตรีมเสียงแล้ว audio/mp4 อาจเป็น ใช้แล้ว. ไม่ว่าในกรณีใด ไฟล์เหล่านี้บ่งชี้ว่าไฟล์เป็นไปตาม "MP4" ข้อกำหนด ISO/IEC 14496-1:2000 รูปแบบไฟล์ระบบ 3 . ข้อควรพิจารณาของ IANA ส่วนนี้อธิบายประเภท MIME และชื่อที่จะใช้กับ เนื้อหา MPEG-4 ต่างๆ ส่วนจาก 4.1 ถึง 4.5 ลงทะเบียนห้าใหม่ ประเภท MIME กับ IANA Lim & Singer Standards Track [หน้า 3] RFC 4337 ประเภท MIME MPEG-4 มีนาคม 2549 3.1 . ไฟล์ MP4 ชื่อประเภทสื่อ MIME: วิดีโอ ชื่อประเภทย่อย MIME: mp4 พารามิเตอร์ที่จำเป็น: none พารามิเตอร์เสริม: none ข้อควรพิจารณาในการเข้ารหัส: โดยทั่วไปแนะนำให้ใช้ base64 IS ไฟล์คือ ไบนารีและควรส่งโดยไม่มีการแปลง CR/LF, 7 บิต การปอก ฯลฯ การพิจารณาความปลอดภัย: ดูมาตรา 5 ของ RFC 4337 ข้อควรพิจารณาในการทำงานร่วมกัน: จำนวนของการทำงานร่วมกัน มีการใช้งานอยู่ภายในชุมชน ISO/IEC 14496 และนั่น ชุมชนมีซอฟต์แวร์อ้างอิงสำหรับอ่านและเขียนไฟล์ รูปแบบ. ข้อกำหนดที่เผยแพร่: ISO/IEC 14496-1:2001 แอปพลิเคชั่น: มัลติมีเดีย ข้อมูลเพิ่มเติม: จำนวนเวทย์มนตร์: none นามสกุลไฟล์: mp4 และ mpg4 มีการประกาศที่ < http://pitch.nist.gov/nics/ >. รหัสประเภทไฟล์ Macintosh: mpg4 ลงทะเบียนกับ Apple แล้ว บุคคลที่ติดต่อเพื่อขอข้อมูล: David Singer, singer@apple.com การใช้งานที่ตั้งใจไว้: ทั่วไป ผู้เขียน/เปลี่ยนตัวควบคุม: David Singer, ISO/IEC 14496 file format เก้าอี้ Lim & Singer Standards Track [หน้า 4] RFC 4337 ประเภท MIME MPEG-4 มีนาคม 2549 3.2 . ไฟล์ MP4 พร้อมเสียง แต่ไม่มีการนำเสนอภาพ ชื่อประเภทสื่อ MIME: เสียง ชื่อประเภทย่อย MIME: mp4 พารามิเตอร์ที่จำเป็น: none พารามิเตอร์เสริม: none ข้อควรพิจารณาในการเข้ารหัส: โดยทั่วไปแนะนำให้ใช้ base64 IS ไฟล์คือ ไบนารีและควรส่งโดยไม่มีการแปลง CR/LF, 7 บิต การปอก ฯลฯ การพิจารณาความปลอดภัย: ดูมาตรา 5 ของ RFC 4337 ข้อควรพิจารณาในการทำงานร่วมกัน: จำนวนของการทำงานร่วมกัน มีการใช้งานอยู่ภายในชุมชน ISO/IEC 14496 และนั่น ชุมชนมีซอฟต์แวร์อ้างอิงสำหรับอ่านและเขียนไฟล์ รูปแบบ. ข้อกำหนดที่เผยแพร่: ISO/IEC 14496-1:2001 แอปพลิเคชั่น: มัลติมีเดีย ข้อมูลเพิ่มเติม: จำนวนเวทย์มนตร์: none นามสกุลไฟล์: mp4 และ mpg4 มีการประกาศที่ < http://pitch.nist.gov/nics/ >. รหัสประเภทไฟล์ Macintosh: mpg4 ลงทะเบียนกับ Apple แล้ว บุคคลที่ติดต่อเพื่อขอข้อมูล: David Singer, singer@apple.com การใช้งานที่ตั้งใจไว้: ทั่วไป ผู้เขียน/เปลี่ยนตัวควบคุม: David Singer, ISO/IEC 14496 file format เก้าอี้. Lim & Singer Standards Track [หน้า 5] RFC 4337 ประเภท MIME MPEG-4 มีนาคม 2549 3.3 . ไฟล์ MP4 ที่มี MPEG-4 System Stream และไม่ใช่ทั้งภาพ และการนำเสนอเสียง ชื่อประเภทสื่อ MIME: แอปพลิเคชัน ชื่อประเภทย่อย MIME: mp4 พารามิเตอร์ที่จำเป็น: none พารามิเตอร์เสริม: none ข้อควรพิจารณาในการเข้ารหัส: โดยทั่วไปแนะนำให้ใช้ base64 IS ไฟล์คือ ไบนารีและควรส่งโดยไม่มีการแปลง CR/LF, 7 บิต การปอก ฯลฯ การพิจารณาความปลอดภัย: ดูมาตรา 5 ของ RFC 4337 ข้อควรพิจารณาในการทำงานร่วมกัน: จำนวนของการทำงานร่วมกัน มีการใช้งานอยู่ภายในชุมชน ISO/IEC 14496 และนั่น ชุมชนมีซอฟต์แวร์อ้างอิงสำหรับอ่านและเขียนไฟล์ รูปแบบ. ข้อกำหนดที่เผยแพร่: ISO/IEC 14496-1:2001 แอปพลิเคชั่น: มัลติมีเดีย ข้อมูลเพิ่มเติม: จำนวนเวทย์มนตร์: none นามสกุลไฟล์: mp4 และ mpg4 มีการประกาศที่ < http://pitch.nist.gov/nics/ >. รหัสประเภทไฟล์ Macintosh: mpg4 ลงทะเบียนกับ Apple แล้ว บุคคลที่ติดต่อเพื่อขอข้อมูล: David Singer, singer@apple.com การใช้งานที่ตั้งใจไว้: ทั่วไป ผู้เขียน/เปลี่ยนตัวควบคุม: David Singer, ISO/IEC 14496 file format เก้าอี้ Lim & Singer Standards Track [Page 6] RFC 4337 MPEG-4 MIME Types March 2006 3.4. Initial Object Descriptor (IOD) in Binary Format MIME media type name: application MIME subtype name: mpeg4-iod Required parameters: none Optional parameters: none Encoding considerations: base64 is generally preferred; files are binary and should be transmitted without CR/LF conversion, 7-bit stripping, etc. Security considerations: See section 5 of RFC 4337. Interoperability considerations: A number of interoperating implementations exist within the ISO/IEC 14496 community, and that community has reference software for reading and writing the file format. Published specification: ISO/IEC 14496-1:2001 Applications: Multimedia Additional information: Magic number(s): none File extension(s): none mp4 and mpg4 are both declared at <http://pitch.nist.gov/nics/>. Macintosh File Type Code(s): mpg4 is registered with Apple. Person to contact for info: David Singer, singer@apple.com Intended usage: Common Author/Change controller: David Singer, ISO/IEC 14496 file format chair Lim & Singer Standards Track [Page 7] RFC 4337 MPEG-4 MIME Types March 2006 3.5. Initial Object Descriptor (IOD) in Textual Format MIME media type name: application MIME subtype name: mpeg4-iod-xmt Required parameters: none Optional parameters: none Encoding considerations: none Security considerations: See section 5 of RFC 4337. Interoperability considerations: A number of interoperating implementations exist within the ISO/IEC 14496 community, and that community has reference software for reading and writing the file format. Published specification: ISO/IEC 14496-1:2001 AMD2. Applications: Multimedia Additional information: Magic number(s): none File extension(s): mp4 and mpg4 are both declared at <http://pitch.nist.gov/nics/>. Macintosh File Type Code(s): mpg4 is registered with Apple. Person to contact for info: David Singer, singer@apple.com Intended usage: Common Author/Change controller: David Singer, ISO/IEC 14496 file format chair Lim & Singer Standards Track [Page 8] RFC 4337 MPEG-4 MIME Types March 2006 4. Security Considerations It is possible to inject non-compliant MPEG streams (Audio, Video, and Systems) in the MP4 file to overload the receiver/decoder's buffers. This might compromise the functionality of the receiver or even crash it. This is especially true for end-to-end systems like MPEG, where the buffer models are precisely defined. An MP4 file supports the storage of stream types, including commands that are executed on the terminal such as OD command and BIFS commands, and programmatic content such as MPEG-J (Java(TM) Byte Code) and ECMASCRIPT. It is possible to use one or more of the above in a manner non-compliant to MPEG to crash the receiver or temporarily make it unavailable. Authentication mechanisms can be used to validate of the sender and the data to prevent security problems due to non-compliant malignant MP4 files. A security model is defined in ISO/IEC 14496 Systems MP4 files containing MPEG-J contents that comprises Java(TM) classes and objects. MPEG-J defines a set of Java(TM) APIs and a secure execution model. MPEG-J content can call this set of APIs and Java(TM) methods from a set of Java packages supported in the receiver within the defined security model. According to this security model, downloaded byte code is forbidden to load libraries, to define native methods, to start programs, to read or write files, or to read system properties. 5. Acknowledgements This document has benefited greatly by contributions from many people, including Mike Coleman, Jean-Claude Duford, Viswanathan Swaminathan, Peter Westerink, Carsten Herpel, Olivier Avaro, Paul Christ, Zvi Lifshitz, and many others. Their insight, foresight, and contribution is gratefully acknowledged. Little has been invented here by the author; this is mostly a collation of greatness that has gone before. 6. Normative References [1] Schulzrinne, H., Casner, S., Frederick, R., and V. Jacobson, "RTP: A Transport Protocol for Real-Time Applications", STD 64, RFC 3550, July 2003. [2] ISO/IEC 14496-1 "Information technology - Coding of audio-visual objects - Part 1 : Systems", 3rd ed. 2004. Lim & Singer Standards Track [Page 9] RFC 4337 MPEG-4 MIME Types March 2006 [3] ISO/IEC 14496-12 "Information technology - Coding of audio- visual objects - Part 12 : ISO Base Media File Format", December 2003. [4] ISO/IEC 14496-14 "Information technology - Coding of audio- visual objects - Part 14 : MP4 File Format", January 2004. [5] ISO/IEC 14496-15 "Information technology - Coding of audio- visual objects - Part 15 : AVC File Format", 2004. [6] ISO/IEC 14496-10:2004 "Information technology -- Coding of audio-visual objects -- Part 10: Advanced Video Coding", 2nd edition, 2004. Authors' Addresses Young-Kwon LIM net&tv Inc. Room 802 Hanseo Building 1582-6 Seocho-3-Dong Seocho-Gu Seoul, 137-875, Korea Phone: +82-2-581-2305 EMail: young@netntv.co.kr David Singer Apple Computer, Inc. One Infinite Loop, MS:302-3MT Cupertino CA 95014 USA Phone: +1 408 974 3162 EMail: singer@apple.com Lim & Singer Standards Track [Page 10] RFC 4337 MPEG-4 MIME Types March 2006 Full Copyright Statement Copyright (C) The Internet Society (2006). This document is subject to the rights, licenses and restrictions contained in BCP 78, and except as set forth therein, the authors retain all their rights. This document and the information contained herein are provided on an "AS IS" basis and THE CONTRIBUTOR, THE ORGANIZATION HE/SHE REPRESENTS OR IS SPONSORED BY (IF ANY), THE INTERNET SOCIETY AND THE INTERNET ENGINEERING TASK FORCE DISCLAIM ALL WARRANTIES, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO ANY WARRANTY THAT THE USE OF THE INFORMATION HEREIN WILL NOT INFRINGE ANY RIGHTS OR ANY IMPLIED WARRANTIES OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Intellectual Property The IETF takes no position regarding the validity or scope of any Intellectual Property Rights or other rights that might be claimed to pertain to the implementation or use of the technology described in this document or the extent to which any license under such rights might or might not be available; nor does it represent that it has made any independent effort to identify any such rights. Information on the procedures with respect to rights in RFC documents can be found in BCP 78 and BCP 79. Copies of IPR disclosures made to the IETF Secretariat and any assurances of licenses to be made available, or the result of an attempt made to obtain a general license or permission for the use of such proprietary rights by implementers or users of this specification can be obtained from the IETF on-line IPR repository at http://www.ietf.org/ipr IETF เชิญชวนผู้มีส่วนได้ส่วนเสียให้ความสนใจใด ๆ ลิขสิทธิ์ สิทธิบัตร หรือคำขอรับสิทธิบัตร หรือกรรมสิทธิ์อื่นๆ สิทธิ์ที่อาจครอบคลุมเทคโนโลยีที่อาจจำเป็นต้องดำเนินการ มาตรฐานนี้ โปรดแจ้งข้อมูลไปยัง IETF ที่ ietf-ipr@ietf.org. การรับทราบ การจัดหาเงินทุนสำหรับฟังก์ชัน RFC Editor นั้นจัดทำโดย IETF กิจกรรมสนับสนุนการบริหาร (IASA) Lim & Singer Standards Track [หน้า 11]

`Facebook.com `